src/github.com/matrix-org/dendrite/clientapi/routing/register.go

```diff
@@ -810,3 +562,32 @@ func RegisterAvailable(
 		},
 	}
 }
+
+// validateApplicationService checks if a provided application service token
```

Comment:
was there a reason to move this to the end of the file rather than leave it at line 319 (https://github.com/matrix-org/dendrite/pull/413/files#diff-789859953fb6c45e704d547fce33b9eeL319) ? Moving it makes it hard to review.